src/TLuaInterpreter.cpp

```diff
@@ -1787,10 +1789,27 @@ int TLuaInterpreter::saveMap(lua_State* L)
         }
     }
 
+    if (lua_gettop(L) == 2) {
+        if (!lua_isstring(L, 1)) {
+            lua_pushstring(L, "saveMap: wrong argument location.");
+            lua_error(L);
+            return 1;
+        } else {
+            location = lua_tostring(L, 1);
+        }
+        if (!lua_isnumber(L, 2)) {
+            lua_pushstring(L, "saveMap: wrong argument version.");
+            lua_error(L);
```

Comment:
:nerd_face:  As it happens, `lua_error(L);` __never__ returns so the `return 1;` is unreachable - to avoid some on-line code checkers moaning about this we are adopting the pattern of, when working on an affected function, replacing:

    lua_error(L);
    return 1:

with:

    return lua_error(L);

which shuts them up and behaves exactly the same...